src/QsCompiler/Tests.Compiler/TestCases/LocalVerification.qs

```diff
@@ -51,6 +51,144 @@ namespace Microsoft.Quantum.Testing.LocalVerification {
         using ((q1, q2, q3) = (Qubit(), (Qubit(), Qubit()))) {}
     }
 
+    operation VariableDeclaration11<'T>(cnt: Int, arg : 'T) : Unit {
+        let recur = VariableDeclaration11; // not allowed
+        recur(cnt - 1, arg);
+    }
+
+    operation VariableDeclaration12<'T>(cnt: Int, arg : 'T) : Unit {
+        let tuple = (1, (VariableDeclaration12, "")); // not allowed
+        let (_, recur) = tuple;
+        recur(cnt - 1, arg);
+    }
+
+    operation VariableDeclaration13<'T>(cnt: Int, arg : 'T) : Unit {
+        mutable arr = new ((Int, 'T) => Unit)[0];
+        set arr += [VariableDeclaration12]; 
+        arr[0](cnt - 1, arg);
+    }
+
+    operation VariableDeclaration14<'T>(cnt: Int, arg : 'T) : Unit {
+        let arr = [VariableDeclaration14];
+        mutable _arr = [VariableDeclaration14];
+        arr[0](cnt - 1, arg);
+    }
+
+    operation VariableDeclaration15<'T>(cnt: Int, arg : 'T) : Unit {
+        mutable arr = new ((Int, 'T) => Unit)[0];
+        set arr = [VariableDeclaration15];
+        arr[0](cnt - 1, arg);
+    }
+
+    operation VariableDeclaration16<'T>(cnt: Int, arg : 'T) : Unit {
+        mutable arr = new ((Int, 'T) => Unit)[0];
+        set arr += [VariableDeclaration16];
+        arr[0](cnt - 1, arg);
+    }
+
+    operation VariableDeclaration17<'T>(cnt: Int, arg : 'T) : Unit {
+        mutable arr = new ((Int, 'T) => Unit)[1];
+        set arr w/= 0 <- VariableDeclaration17;
+        arr[0](cnt - 1, arg);
+    }
+
+    operation VariableDeclaration18<'T>(cnt: Int, arg : 'T) : Unit {
+        mutable arr = new ((Int, 'T) => Unit)[2];
+        set arr w/= 0 .. 1 <- [VariableDeclaration18, VariableDeclaration17];
```

Comment:
```suggestion
        set arr w/= 0 .. 1 <- [VariableDeclaration18, VariableDeclaration18];
```
Was this meant to be a recursive reference also?

Reply:
same here.